pkg/ext/store.go

```diff
@@ -2,92 +2,260 @@ package ext
+// (and Rancher UI) to display a table of the items.
 //
-// Errors returned by the Store should use errors from k8s.io/apimachinery/pkg/api/errors. This
-// will ensure that the right error will be returned to the clients (eg: kubectl, client-go) so
-// they can react accordingly. For example, if an object is not found, store should
-// return the following error:
+// Currently, we use the default table convertor which will show two columns: Name and Created At.
+func ConvertToTable[T runtime.Object](ctx context.Context, object runtime.Object, tableOptions runtime.Object, groupResource schema.GroupResource, columnDefs []metav1.TableColumnDefinition, convertFn ConvertFunc[T]) (*metav1.Table, error) {
+	result, err := convertToTable(ctx, object, tableOptions, groupResource, columnDefs, convertFn)
+	if err != nil {
+		return nil, convertError(err)
+	}
+	return result, nil
+}
+
+func ConvertToTableDefault[T runtime.Object](ctx context.Context, object runtime.Object, tableOptions runtime.Object, groupResource schema.GroupResource) (*metav1.Table, error) {
+	return ConvertToTable[T](ctx, object, tableOptions, groupResource, nil, nil)
+}
+
+func convertToTable[T runtime.Object](ctx context.Context, object runtime.Object, tableOptions runtime.Object, groupResource schema.GroupResource, columnDefs []metav1.TableColumnDefinition, convertFn ConvertFunc[T]) (*metav1.Table, error) {
+	defaultTableConverter := rest.NewDefaultTableConvertor(groupResource)
+	table, err := defaultTableConverter.ConvertToTable(ctx, object, tableOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if columnDefs == nil {
+		return table, nil
+	}
+
+	// Override only if there were definitions before (to respect the NoHeader option)
+	if len(table.ColumnDefinitions) > 0 {
+		table.ColumnDefinitions = columnDefs
+	}
+	table.Rows = []metav1.TableRow{}
+	fn := func(obj runtime.Object) error {
+		objT, ok := obj.(T)
+		if !ok {
+			var zeroT T
+			return fmt.Errorf("expected %T but got %T", zeroT, obj)
+		}
+		cells := convertFn(objT)
+		if len(cells) != len(columnDefs) {
+			return fmt.Errorf("defined %d columns but got %d cells", len(columnDefs), len(cells))
+		}
+
+		table.Rows = append(table.Rows, metav1.TableRow{
+			Cells:  cellStringToCellAny(cells),
+			Object: runtime.RawExtension{Object: obj},
+		})
+		return nil
+	}
+	switch {
+	case meta.IsListType(object):
+		if err := meta.EachListItem(object, fn); err != nil {
+			return nil, err
+		}
+	default:
+		if err := fn(object); err != nil {
+			return nil, err
+		}
+	}
+
+	return table, nil
+}
+
+func cellStringToCellAny(cells []string) []any {
+	res := []any{}
+	for _, cell := range cells {
+		res = append(res, cell)
+	}
+	return res
+}
+
+// CreateOrUpdate helps implement [rest.Updater] by handling most of the logic.
 //
-//	apierrors.NewNotFound(ctx.GroupVersionResource.GroupResource(), name)
+// createValidation is used to do some validation on the object before creating
```

Comment:
```suggestion
// createValidation is used to do validation on the object before creating
```

Reply:
Reworded those comments so you can have a look at them again.